src/app/data.service.ts

```diff
@@ -4,69 +4,42 @@ import { JobFamily } from './JobFamily';
 import { Capability } from './Capability';
 import { Band } from './Band';
 import { Role } from './Role';
+import { Observable } from 'rxjs';
 
 
 @Injectable({
   providedIn: 'root'
 })
 export class DataService {
 
-  jobFamilies: JobFamily[];
-  capabilities: Capability[];
-  bands: Band[];
-  roles: Role[];
-  role: Role = new Role();
+  constructor(private http: HttpClient) { 
+    this.getEverything();
+  }
 
-  constructor(private http: HttpClient) { }
+  public getEverything(){
+    this.getAllJobFamilies();
+    this.getAllCapabilities();
+    this.getAllBands();
+    this.getAllRoles();
+  }
 
-  
-  getAllJobFamilies() : void {
-    this.http.get<JobFamily[]>('/api/jobfamilies').subscribe(res => {
-      console.log(res)
-      if(res[0] == null){
-        console.error(res);
-      } else {
-        this.jobFamilies = res;
-      }
-    });
+  public getAllJobFamilies() : Observable<JobFamily[]> {
+    return this.http.get<JobFamily[]>('/api/families');
   }
 
-  getAllCapabilities() : void {
-    this.http.get<Capability[]>('/api/capabilities').subscribe(res => {
-      console.log(res)
-      if(res[0] == null){
-        console.error(res);
-      } else {
-        this.capabilities = res;
-      }
-    });
+  public getAllCapabilities() : Observable<Capability[]> {
+    return this.http.get<Capability[]>('/api/capabilities');
   }
 
-  getAllBands() : void {
-    this.http.get<Band[]>('/api/bands').subscribe(res => {
-      console.log(res)
-      if(res[0] == null){
-        console.error(res);
-      } else {
-        this.bands = res;
-      }
-    });
+  public getAllBands() : Observable<Band[]> {
+    return this.http.get<Band[]>('/api/bands');
   }
 
-  getAllRoles() : void {
-    this.http.get<Role[]>('/api/roles').subscribe(res => {
-      console.log(res)
-      if(res[0] == null){
-        console.error(res);
-      } else {
-        this.roles = res;
-      }
-    });
+  public getAllRoles() : Observable<Role[]> {
+    return this.http.get<Role[]>('/api/roles');
   }
 
-  getRole(id) : void {
-    this.http.get<Role>('/api/role/' + id).subscribe(role => {
-      this.role = role;
-    });
+  public getRole(id) : Observable<JobFamily[]> {
+    return this.http.get<JobFamily[]>('/api/role' + id);
```

Comment:
Would be good to delimit the role and ID for a role with a ? or an & symbol, rather than having /api/role2 for example.